src/GdbCommandHandler.cc

```diff
@@ -105,6 +107,31 @@ class RRCmd(gdb.Command):
         response = gdb_unescape(rv_match.group(1))
         gdb.write(response)
 
+
+class RRRunC(gdb.Command):
+  """Helper to run code for different language environments always in c"""
+  def __init__(self):
+    gdb.Command.__init__(self, 'rr-run-c',
+                       gdb.COMMAND_USER, gdb.COMPLETE_NONE, False)
+
+  def invoke(self, arg, from_tty):
+    # sorry, the whole parsing function is not very robust
+    # not sure how to improve: maybe gdb spec/see source if this is ok
+    # The current source language is 
+    output = gdb.execute('show lang', False, True)
+    start = output.find('"')
+    end = output.find('"', start + 1)
+    user_lang = output[start + 1 : end]
+    if user_lang.startswith('auto;'): # auto currently name
+      user_lang = user_lang[user_lang.rfind(' ') : ]
+    if arg and arg[0] == '"' and arg[-1] == '"':
+      arg = arg[1:-1].replace('\\n', '\n')
+    gdb.execute('set lang c')
+    gdb.execute(arg)
+    gdb.execute('set lang %s' % user_lang)
```

Comment:
If the current language is `auto` then this sets it to whatever the auto language is, e.g. `c`. Instead, we should do `set lang auto` again.

Reply:
yeah !